cli/src/commands/git/fetch.rs

```diff
@@ -108,16 +108,25 @@ pub fn cmd_git_fetch(
         }
     }
 
-    match &unmatched_patterns[..] {
-        [] => {} // Everything matched, all good
-        [pattern] if pattern.is_exact() => {
-            return Err(user_error(format!("No git remote named '{pattern}'")))
-        }
-        patterns => {
-            return Err(user_error(format!(
-                "No matching git remotes for patterns: {}",
-                patterns.iter().join(", ")
-            )))
+    match (!matching_remotes.is_empty(), &unmatched_patterns[..]) {
```

Comment:
Still fairly new to rust - open to suggestions on (more) idiomatic patterns here.